hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/customfees/CustomFeeAssessor.java

```diff
@@ -131,13 +133,24 @@ private void revalidateAssessmentResult(
             final var entryValue = entry.getValue();
             for (final var entryTx : entryValue.entrySet()) {
                 final Long htsBalanceChange = entryTx.getValue();
+                final var accountId = entryTx.getKey();
+                final var tokenRel = tokenRelStore.get(accountId, entry.getKey());
+                // revalidate collector's association
+                if (htsBalanceChange > 0) {
+                    if (tokenRel == null) {
+                        final var currentAccount = accountStore.getAccountById(accountId);
+                        final var mayBeAutoAssociatedHere = currentAccount != null
+                                && (currentAccount.maxAutoAssociations() > currentAccount.usedAutoAssociations()
+                                        || currentAccount.maxAutoAssociations() == -1);
+                        validateTrue(mayBeAutoAssociatedHere, TOKEN_NOT_ASSOCIATED_TO_FEE_COLLECTOR);
```

Comment:
I guess this wouldn't hurt. But why do we need extra check ?

Reply:
Just to differentiate between `TOKEN_NOT_ASSOCIATED_TO_ACCOUNT` and `TOKEN_NOT_ASSOCIATED_TO_FEE_COLLECTOR` ?